samcli/commands/_utils/custom_options/hook_package_id_option.py

```diff
@@ -46,6 +50,20 @@ def handle_parse_result(self, ctx, opts, args):
                     raise click.BadParameter(
                         f"Parameters {opt_name}, and {','.join(self._invalid_coexist_options)} can not be used together"
                     )
+            # check beta-feature
+            beta_features = opts.get("beta_features")
+            if beta_features is None:
+                terraform_support_message = (
+                    "Supporting Terraform applications is a beta feature.\n"
```

Comment:
hook option are a general IaC prepare hook right? we may need a way to figure out its terraform features that are being enabled and not some other Iac?

Reply:
for now, we only have Terraform supported, and we are going to go GA and remove beta features flag before adding another IaC support. I think we can handle this scenario when we add a new IaC support, or start accepting external hook packages.